packages/react-server/src/logger/logger.ts

```diff
@@ -1,50 +1,87 @@
+      this.logger.log(`${PREFIX}${message}`);
+    } else {
+      this.buffer = `${this.buffer}\n${message}`;
+    }
+  }
 }
 
-export enum Verbosity {
-  Off,
-  Debug,
+function initialRequestMessage(request: Request): string {
+  const requestMethod = `${request.method.toUpperCase()} "${request.url}"`;
+  return `Started ${requestMethod} for at ${new Date().toISOString()}`;
 }
 
-export class Logger {
-  private logger: Pick<Console, 'log' | 'error'> =
-    this.options.level === Verbosity.Off ? noopLogger : console;
+function endRequestMessage(ctx: Context, requestDuration: number): string {
+  const httpStatus = `${ctx.status} ${ctx.message || ''}`;
+  const duration = `${requestDuration.toFixed(0)}ms`;
 
-  constructor(private options: LoggerOptions = {level: Verbosity.Off}) {}
+  return `Completed ${httpStatus} at ${new Date().toISOString()} in ${duration}`;
+}
 
-  log(message: string) {
-    this.logger.log(`${PREFIX}${message}`);
-  }
+function endRequest(ctx: Context, requestDuration: number) {
+  ctx.state.logger.log(endRequestMessage(ctx, requestDuration));
 
-  error(message: string) {
-    this.logger.error(`${PREFIX}${message}`);
+  if (process.env.NODE_ENV === 'development') {
+    return;
   }
-}
 
-export function getLogger(ctx: Context): Logger {
-  return ctx.state[LOGGER];
+  /* eslint-disable babel/camelcase */
+  const logObject: any = {
+    datetime: new Date().toISOString(),
+    http_method: ctx.method.toUpperCase(),
+    http_response: ctx.message || '',
+    http_status: ctx.status,
+    uri: ctx.originalUrl,
+    user_agent: ctx.header['User-Agent'],
+    payload: ctx.state.logger.buffer,
+  };
+  /* eslint-enable babel/camelcase */
+
+  // eslint-disable-next-line no-console
+  console.log(
+    JSON.stringify(
+      logObject,
+      undefined,
+      process.env.NODE_ENV === 'production' ? undefined : 2,
+    ),
+  );
 }
 
-export function setLogger(ctx: Context, logger: Logger) {
-  ctx.state[LOGGER] = logger;
+function requestDuration(requestStartTime: [number, number]) {
+  const duration = process.hrtime(requestStartTime);
+  const ms = duration[0] * 1000 + duration[1] / 1e6;
+  return Math.round(ms);
 }
 
-export function createLogger(options?: LoggerOptions) {
-  const logger = new Logger(options);
+export async function requestLogger(ctx: Context, next: KoaNextFunction) {
+  const requestStartTime = process.hrtime();
 
-  return async function loggerMiddleware(ctx: Context, next: KoaNextFunction) {
-    setLogger(ctx, logger);
+  ctx.state.logger = new Logger();
+  ctx.state.logger.log(initialRequestMessage(ctx.request));
 
+  try {
     await next();
-  };
+  } catch (error) {
+    ctx.state.logger.log('Error during server execution, see details below.');
+    ctx.state.logger.log(
+      `${error.stack || error.message || 'No stack trace was present'}`,
+    );
+  } finally {
+    endRequest(ctx, requestDuration(requestStartTime));
+  }
 }
 
 export const noopLogger = {
```

Comment:
noticed you removed this below, wondering it is still needed?

Reply:
good catch, not needed anymore either